demos/common/mqtt_subscription_manager/mqtt_subscription_manager.c

```diff
@@ -0,0 +1,171 @@
+/*
+ * FreeRTOS V202011.00
+ * Copyright (C) 2020 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * https://www.FreeRTOS.org
+ * https://aws.amazon.com/freertos
+ *
+ */
+
+/**
+ * @file subscription_manager.c
+ * @brief Functions for managing MQTT subscriptions.
+ */
+
+/* Standard includes. */
+#include <string.h>
+
+/* Subscription manager header include. */
+#include "mqtt_subscription_manager.h"
+
+
+bool SubscriptionManager_AddSubscription( SubscriptionElement_t * pxSubscriptionList,
```

Comment:
These names do not match the header file. Also, these files are largely duplicates of `demos/coreMQTT_Agent/subscription_manager.c`/`.h`. It might be better just to move those files

Reply:
I believe the common folder is a better place to keep the `subscription_manager`. The component can be removed from the `coreMQTT-Agent` and reused from the common folder.
Updated header in https://github.com/aws/amazon-freertos/pull/3188/commits/450fda5aa67e04ef89b710317773948af5a9259c

Reply:
I agree to Shubham's comment , the demos/common seems to be better instead of demo/coreMQTT_Agent for the subscription manager